metadata-ingestion/tests/unit/schema/test_json_schema_util.py

```diff
@@ -671,3 +684,28 @@ def test_simple_object():
     ]
     assert_field_paths_match(fields, expected_field_paths)
     assert isinstance(fields[0].type.type, RecordTypeClass)
+
+
+def test_required_field():
+    schema = {
+        "$id": "test",
+        "$schema": "http://json-schema.org/draft-06/schema#",
+        "properties": {
+            "a_str": {
+                "description": "Example String",
+                "type": "string",
+            },
+            "b_str": {"type": ["string", "null"]},
+        },
+        "required": ["b_str"],
+    }
+    fields = list(JsonSchemaTranslator.get_fields_from_schema(schema))
+    expected_field_paths: List[str] = [
+        "[version=2.0].[type=object].[type=string].a_str",
+        "[version=2.0].[type=object].[type=string].b_str",
+    ]
+    assert_field_paths_match(fields, expected_field_paths)
+    assert_fields_are_valid(fields)
+    assert fields[0].nullable is False
```

Comment:
Would also be good to assert the other field is nullable

Reply:
added